src/components/Select/Select.tsx

```diff
@@ -535,7 +536,9 @@ export const Select: FC<SelectProps> = React.forwardRef(
             onVisibleChange={(isVisible) => setDropdownVisibility(isVisible)}
             overlay={isLoading ? spinner : <OptionMenu options={options} />}
             showDropdown={showDropdown}
-            visible={dropdownVisible}
+            visible={
+              dropdownVisible && (showEmptyDropdown || options.length > 0)
```

Comment:
Does this account for the type-ahead scenario and still show the 'No match found' result? Never showing makes sense for on click with no input, but should provide some information with a partially input text.

Reply:
You are right, this is done to handle the case of `click with no input`, and the `No match found` message will be shown with partially input text.